doc/build-helpers/testers.chapter.md

```diff
@@ -169,15 +169,24 @@ The build will fail if `shellcheck` finds any issues.
 
 Checks that the output from running a command contains the specified version string in it as a whole word.
 
-NOTE: In most cases, [`versionCheckHook`](#versioncheckhook) should be preferred, but this function is provided and documented here anyway. The motivation for adding either tests would be:
+Note: Given the current limitations of Nixpkgs's tooling and CI, `versionCheckHook` fits better than `testers.testVersion` in most of typical situations.
 
-- Catch dynamic linking errors and such and missing environment variables that should be added by wrapping.
-- Probable protection against accidentally building the wrong version, for example when using an "old" hash in a fixed-output derivation.
+Although being very simple and naive, this test is useful:
+
+- It assures the main program can run.
+- It brings awareness to dynamic linking errors, missing or malformed environment variables and other runtime errors.
+- It provides a minor protection against fixed-output hash errors that usually happen when forgetting to update hashes.
+
+By default, the command to be run will be inferred from the given `package` attribute: it will check `meta.mainProgram` first, falling back to `pname` or `name`.
 
-By default, the command to be run will be inferred from the given `package` attribute:
-it will check `meta.mainProgram` first, and fall back to `pname` or `name`.
 The default argument to the command is `--version`, and the version to be checked will be inferred from the given `package` attribute as well.
 
+Note: see also:
+
+- [`versionCheckHook`](#versioncheckhook), an independent hook with similar functionality.
+  - [A comparison between both](#versioncheckhook-comparison-testversion).
+  - Given the current limitations of Nixpkgs's tooling and CI, `versionCheckHook` fits better than `testers.testVersion` in most of typical situations.
```

Comment:
You already wrote about the limitations above.

Reply:
You required to put it on different places, usually before a section to save time and after them to summarize conclusions.

Reply:
> You required to put it on different places, usually before a section to save time and after them to summarize conclusions.

It is correct that I stated that there is no choice but to mention it both `passthru.chapter.md` and `testers.chapter.md`, but I think it should be mentioned in the same file twice, in less then 10 lines distance between each mention.

Reply:
I am failing to understand your requirement here.

You said while commenting this same file `doc/build-helpers/testers.chapter.md`:

> [Would you agree to (after all) write here the bottom line of the comparison? Which is that versionCheckHook fits most typical cases?](https://github.com/NixOS/nixpkgs/pull/348025#discussion_r1826913885)

And later _in the same file_ `doc/build-helpers/testers.chapter.md`:

> [In fact it would have been better to state the bottom line of the comparison here to save people time when reading.](https://github.com/NixOS/nixpkgs/pull/348025#discussion_r1826913985)

Now I don't know if you are [complaining about the duplication](https://github.com/NixOS/nixpkgs/pull/348025#discussion_r1827407783) _in the same file_ or if you are [agreeing with this duplication](https://github.com/NixOS/nixpkgs/pull/348025#discussion_r1827480362) by stating

> . . . but I think it should be mentioned in the same file twice, in less then 10 lines distance between each mention.

IMHO I have no problem duplicating this.

(BTW the duplication around multiple files is not relevant here.)

B

Reply:
> > . . . but I think it should be mentioned in the same file twice, in less then 10 lines distance between each mention.
> 
> IMHO I have no problem duplicating this.

Oh I'm so sorry @AndersonTorres . I forgot to negate that sentence, I think it _shouldn't_ be mentioned in the same file twice, in a such a close distance of <10 lines apart.

Reply:
Done.